### PR TITLE
Log scrape logs to get docker images used for building

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade/issues/10247

### CODE REVIEW

File: src/Telemetry/Microsoft.DotNet.AzureDevOpsTimelineTests/BuildLogScraperTests.cs [hunk 1]
```diff
@@ -84,6 +84,21 @@ public async Task BuildLogScraperShouldExtractOneESHostedPoolImageName()
             Assert.AreEqual(MockAzureClient.MicrosoftHostedAgentImageName, imageName);
         }
 
+        [Test]
+        public async Task BuildLogScraperShouldExtractDockerImageName()
+        {
+            CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
```
Comment (premun) [inline src/Telemetry/Microsoft.DotNet.AzureDevOpsTimelineTests/BuildLogScraperTests.cs:90]: No need to create a source. You can create a token directly or you can use `CancellationToken.None`

File: src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs [hunk 2]
```diff
@@ -162,7 +162,7 @@ public async Task<WorkItem> CreateRcaWorkItem(string project, string title, Canc
 
         public async Task<string> TryGetImageName(
             string logUri,
-            Regex imageNameRegex,
+            Func<string, string> regexFunction,
```
[the remark this answers is not in the document]
Comment (dkurepa) [reply src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs]: that works, replaced

File: src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs [hunk 3]
```diff
@@ -367,10 +367,18 @@ private async Task GetImageNames(List<AugmentedTimelineRecord> records, Cancella
 
             foreach (var record in records)
             {
-                if (!string.IsNullOrEmpty(record.Raw.Log?.Url) && record.Raw.Name == "Initialize job")
+                if (!string.IsNullOrEmpty(record.Raw.Log?.Url))
```
[the remark this answers is not in the document]
Comment (dkurepa) [reply src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs]: Implemented, I guess the last continue isn't needed, but I added it so it looks nicer

File: src/Telemetry/Microsoft.DotNet.AzureDevOpsTimelineTests/BuildLogScraperTests.cs (hunk 1 above)
Comment (dkurepa) [reply src/Telemetry/Microsoft.DotNet.AzureDevOpsTimelineTests/BuildLogScraperTests.cs]: Replaced the token with `CancellationToken.None` everywhere

**Review by premun — APPROVED** [review]

**Pull request merged**